### PR TITLE
Change logic for updating .NET.

### PR DESCRIPTION
**Pull request by JamesWTruher**
also include a message when it looks like a new .NET is not going to be used.

# PR Summary

The workflow to update .NET is sometimes difficult to debug from the log files. This PR attempt to make it more obvious when we are not going to update .net. Also the workflow should provide a better indication of what file should be inspected for issues.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: tools/UpdateDotnetRuntime.ps1 [hunk 1]
```diff
@@ -356,3 +361,6 @@ if ($dotnetUpdate.ShouldUpdate) {
 
     Update-DevContainer
 }
+else {
+    Write-Verbose -Verbose -Message $dotnetUpdate.Message
```
Comment (PaulHigin) [inline tools/UpdateDotnetRuntime.ps1:365]:
```suggestion
    Write-Verbose -Verbose -Message "No update needed.  DotNet update message: $($dotnetUpdate.Message)"
```

**Review by PaulHigin — APPROVED** [review]

File: tools/UpdateDotnetRuntime.ps1 [hunk 2]
```diff
@@ -189,36 +189,41 @@ function Get-DotnetUpdate {
...
-        if ($latestSDKversion -gt $currentVersion) {
+        if ($latestSDKversion -gt $currentVersion -and $null -ne $latestSDKversion.PreReleaseLabel) {
             $shouldUpdate = $true
             $newVersion = $latestSDKversion
         } else {
             $shouldUpdate = $false
-            $newVersion = $null
+            $newVersion = $latestSDKVersionString
+            if ($null -eq $currentVersion.PreReleaseLabel) {
+                $Message = "$latestSDKversion is not preview, update manually."
```
Comment (daxian-dbw) [inline tools/UpdateDotnetRuntime.ps1:225]: Why do we have to update manually for stable SDK versions?

File: tools/UpdateDotnetRuntime.ps1 (hunk 1 above)
Comment (JamesWTruher) [reply tools/UpdateDotnetRuntime.ps1:365]: an update may actually be needed, so i wouldn't want to say "update needed" when the update message says "update manually"

File: tools/UpdateDotnetRuntime.ps1 (hunk 2 above)
Comment (JamesWTruher) [reply tools/UpdateDotnetRuntime.ps1:225]: this was based on a conversation i had with @adityapatwardhan

**Review by adityapatwardhan — APPROVED** [review]

File: tools/UpdateDotnetRuntime.ps1 (hunk 2 above)
Comment (adityapatwardhan) [reply tools/UpdateDotnetRuntime.ps1:225]: @daxian-dbw For stable releases, we dont always update as sometime the update is not applicable to us. For example 6.0.101 is not applicable to us.

Comment (daxian-dbw) [reply tools/UpdateDotnetRuntime.ps1:225]: Thanks for the clarification!

**Pull request merged**